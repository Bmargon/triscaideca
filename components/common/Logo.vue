<template>
  <div class="logo" :style="getStyles">
    <n-link :to="link">
      <img :src="getLogoImage" :alt="altText" />
    </n-link>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'Logo',

  props: {
    width: {
      type: String,
      default: '50px',
    },
    link: {
      type: String,
      default: '/',
    },
    logo: {
      type: String,
      default: 'triscaideca-black-logo.svg',
    },
    altText: {
      type: String,
      default: 'Triscaideca Logo',
    },
  },

  computed: {
    getStyles(): string {
      return `width: ${this.width}`
    },

    getLogoImage(): string {
      return require(`@/assets/images/${this.logo}`)
    },
  },
})
</script>

<style scoped lang="scss"></style>
