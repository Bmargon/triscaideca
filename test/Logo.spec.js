import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Logo from '@/components/common/Logo.vue'

describe('Logo', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(Logo, {
      stubs: {
        NLink: RouterLinkStub,
        NuxtLink: RouterLinkStub,
      },
    })
    expect(wrapper.vm).toBeTruthy()
  })
  test('matches with snapshot', () => {
    const wrapper = shallowMount(Logo, {
      stubs: {
        NLink: RouterLinkStub,
        NuxtLink: RouterLinkStub,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
