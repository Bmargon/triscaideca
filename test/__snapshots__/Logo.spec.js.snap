// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo matches with snapshot 1`] = `
<div
  class="logo"
  style="width: 50px;"
>
  <a>
    <img
      alt="Triscaideca Logo"
      src="[object Object]"
    />
  </a>
</div>
`;
